src/xb-builder-source.c

```diff
@@ -438,6 +438,7 @@ xb_builder_source_get_istream(XbBuilderSource *self, GCancellable *cancellable,
 	XbBuilderSourcePrivate *priv = GET_PRIVATE(self);
 	g_autofree gchar *basename = NULL;
 	GFile *file;
+	GInputStream *istream;
```

Comment:
maybe do g_autoptr(GInputStream) istream = NULL; here and then return g_steal_pointer(&istream) -- otherwise you'll leak this on error.

Reply:
Ouch, you are right, it can leak otherwise. I'll update the merge request.

Reply:
If that passes all the tests then I think we've got something good -- would you need a new release/tarball?

Reply:
I just updated the merge request.

I'm not sure about a new release, it was the first time I saw it, thus nothing common. Feel free to wait few weeks, maybe more fixes will pile up. In any case, I'll left it up to you.